openedx/core/djangoapps/course_roles/models.py

```diff
@@ -0,0 +1,113 @@
+"""
+Models for course roles schema
+"""
+from django.conf import settings
+from django.db import models
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from organizations.models import Organization
+
+
+class Role(models.Model):
+    """
+    Model for a role.
+
+    A role is a collection of permissions that can be assigned to a user.
+    The services field defines in which service UI the role is intended to be assigned, such as CMS and/or LMS.
+
+    """
+    name = models.CharField(max_length=255)
```

Comment:
Carrying over from [a comment in the other PR](https://github.com/openedx/edx-platform/pull/33609#discussion_r1394456032):

> @ormsbee
> Should this have a unique constraint on it?
>
> @julianpalmerio
> We do not need unique names

I don't understand. All the role lookups are done by name, aren't they? Won't the code just assume the names are unique?

Reply:
From a technical point of view the role names do not need to be unique since all checks are based on permissions a user has and the user is linked to the role by id and not name, but from a user point of view having multiple roles with the same name might cause confusion. 

Requiring the name to be unique would cause added complexity if we want to allow custom roles by organization (not only by instance). This will be determined from a use case point of view before a code change is decided upon.

Reply:
How about we add the unique constraint for now, just to make sure the database doesn't accidentally get into a weird state? If/when we decide to add org-specific roles, we can drop the constraint as part of that work–it'd just be a one line change to the model and a migration at that point. It's easier to go from more restrictive to less restrictive than the other way around.